packages/builder/src/components/backend/TableNavigator/ExistingTableDataImport.svelte

```diff
@@ -185,20 +185,22 @@
       </div>
     {/each}
   </div>
-  {#if tableType === DB_TYPE_INTERNAL}
-    <br />
-    <Toggle
-      bind:value={updateExistingRows}
-      on:change={() => (identifierFields = [])}
-      thin
-      text="Update existing rows"
-    />
-    {#if updateExistingRows}
+  <br />
+  <Toggle
```

Comment:
@aptkingston I'd like to hide this toggle when the datasource is MSSQL, but I'm not sure how to best do that. I don't want to call out to the datasources API because I can see in the Chrome network tab that we already fetch the datasources, so I'm guessing there's some way to get at that data but I just don't know how. Halp? 😇

Reply:
The datasources will be available at all times in components like this in the stores, you can get the datasource by its ID using:
```
import { datasources } from "stores/builder"
```

from here you can just get the datasource using the svelte store fetch function `$` - like:
```
const datasourceList = $datasources.list
```

You can then search this list by `_id` to find the datasource this is occurring for, the store also contains the current datasourceId under `$datasources.selectedDatasourceId` - if this is set then it is an external datasource and you can pick it out of the list, then check its sourceType!

Reply:
You're correct - we do indeed already have the required info. You'd achieve this by doing:
```js
import { tables, datasources } from "stores/builder"

export let tableId // already have this prop

$: table = $tables.list.find(table => table._id === tableId)
$: datasource = $datasources.list.find(ds => ds._id === table?.sourceId)
```

Reply:
Thank you very much, gentlemen 🙏